### DISCUSSION TITLE
Consultas sobre el grabado de un postsave

### DISCUSSION THREAD

**Discussion by NicolasPLG** [Cotalker/documentation#176] [2. Questions and Answers] [answered]
Hola 
El tema de referencia es: https://github.com/Cotalker/documentation/discussions/83

Hoy vi con @grimlin31 el postsave en dev apuntando a prod y vi que funcionó de lo más bien, pero me surgió una duda durante el QA.

El valor calculado automáticamente quedará registrado en la answers o será solo visual?. Por qué lo pregunto, porque el objetivo del cálculo automático para el cliente, es que se pueda descargar el archivo y dejar como respaldo del usuario o paciente en este cliente.

Adjunto del ss cuando sale el campo automático.
![image](https://user-images.githubusercontent.com/61419471/118005448-6630e700-b318-11eb-8e33-8ba0dca665ce.png)

Gracias :)

**Comment by rafaelwalter** [comment, marked as answer]
@NicolasPLG postsave es solamente visual. No se guarda en la DB. Para que se guarde, recomiendo que ocupen otro ciclo de vida de questionExec, como onvalidate, para que se calcule el IMC antes de enviar el survey. Al enviar quedará guardado en la DB el valor calculado!

**Reply by NicolasPLG** [reply]
pero para eso sería prácticamente usar el mismo código del postsave antes?.

Algo así
```   "exec" : {
"onvalidate" : {
            "src" : "function run() {const commands  = [] ;const peso = Number(context['responses#peso_hepatopancreaticobiliar']);console.log('peso:', peso); const altura = Number(context['responses#talla_hepatopancreaticobiliar']);console.log('altura:', altura);var imc = peso / (altura**2);console.log('imc:', imc);commands.push( {         cmd: 'SET_RESPONSES',   value: imc  });  commands.push(  {  cmd: 'SET_READONLY',    value: 'true' }); console.log(commands); return commands;}", 
            "context" : "responses#peso_hepatopancreaticobiliar,responses#talla_hepatopancreaticobiliar"
        }
    },
        "postsave" : {
            "src" : "function run() {const commands  = [] ;const peso = Number(context['responses#peso_hepatopancreaticobiliar']);console.log('peso:', peso); const altura = Number(context['responses#talla_hepatopancreaticobiliar']);console.log('altura:', altura);var imc = peso / (altura**2);console.log('imc:', imc);commands.push( {         cmd: 'SET_RESPONSES',   value: imc  });  commands.push(  {  cmd: 'SET_READONLY',    value: 'true' }); console.log(commands); return commands;}", 
            "context" : "responses#peso_hepatopancreaticobiliar,responses#talla_hepatopancreaticobiliar"
        }
    }
```

**Reply by rafaelwalter** [reply]
@NicolasPLG si, el mismo codigo fuente deberia funcionar. Y ahora puedes borrar el `postsave` y tener solamente el `onvalidate`

**Reply by NicolasPLG** [reply]
Entiendo @rafaelwalter realizaré la prueba usando onvalidate en prod, para ver si funciona y luego lo veo en dev.

Cerraré este tema

**Reply by NicolasPLG** [reply]
@rafaelwalter funciona :D, pero era validate y no onvalidate xd.

Cierro aquí y tamben a la espera del postsave producción

**Reply by rafaelwalter** [reply]
increible, gracias @NicolasPLG
